presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -386,20 +404,128 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+                            .map(value -> highLowValueAsPrestoType(value, prestoType))
+                            .collect(toImmutableList());
+                }
             }
+
+            Comparator<Object> comparator = (leftValue, rightValue) -> {
+                Block leftBlock = Utils.nativeValueToBlock(prestoType, leftValue);
+                Block rightBlock = Utils.nativeValueToBlock(prestoType, rightValue);
+                return prestoType.compareTo(leftBlock, 0, rightBlock, 0);
+            };
+            rangeStatistics.setLowValue(lowValueCandidates.stream().min(comparator));
+            rangeStatistics.setHighValue(highValueCandidates.stream().max(comparator));
+
             columnStatistics.addRange(rangeStatistics.build());
             tableStatistics.setColumnStatistics(hiveColumnHandle, columnStatistics.build());
         }
         return tableStatistics.build();
     }
 
+    private boolean isLowHighSupportedForType(Type type)
+    {
+        if (type instanceof DecimalType) {
+            return true;
+        }
+        if (type.equals(TINYINT)
+                || type.equals(SMALLINT)
+                || type.equals(INTEGER)
+                || type.equals(BIGINT)
+                || type.equals(REAL)
+                || type.equals(DOUBLE)
+                || type.equals(DATE)
+                || type.equals(TIMESTAMP)) {
+            return true;
+        }
+        return false;
+    }
+
+    private Object highLowValueAsPrestoType(Object value, Type prestoType)
+    {
+        checkArgument(isLowHighSupportedForType(prestoType), "Unsupported type " + prestoType);
+        requireNonNull(value, "high/low value connot be null");
+
+        if (prestoType.equals(BIGINT)
+                || prestoType.equals(INTEGER)
+                || prestoType.equals(SMALLINT)
+                || prestoType.equals(TINYINT)) {
+            checkArgument(value instanceof Long, "expected Long value but got " + value.getClass());
+            return value;
+        }
+        else if (prestoType.equals(DOUBLE)) {
+            checkArgument(value instanceof Double, "expected Double value but got " + value.getClass());
+            return value;
+        }
+        else if (prestoType.equals(REAL)) {
+            checkArgument(value instanceof Double, "expected Double value but got " + value.getClass());
+            return floatToRawIntBits((float) (double) value);
+        }
+        else if (prestoType.equals(DATE)) {
+            checkArgument(value instanceof LocalDate, "expected LocalDate value but got " + value.getClass());
+            return ((LocalDate) value).toEpochDay();
+        }
+        else if (prestoType.equals(TIMESTAMP)) {
+            checkArgument(value instanceof Long, "expected Long value but got " + value.getClass());
+            long utcMillis = timeZone.convertLocalToUTC((long) value * 1000, false);
+            return utcMillis;
+        }
+        else if (prestoType instanceof DecimalType) {
```

Comment:
I am not sure how relevant it is, but `com.facebook.presto.hive.ColumnarTextHiveRecordCursor#parseDecimalColumn(int, byte[], int, int)` uses `com.facebook.presto.hive.HiveDecimalParser#parseHiveDecimal` which does rounding